# P9 : Produisez une étude de marché

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from matplotlib.collections import LineCollection
import seaborn as sns

## Exploration et nettoyage des données
### Table dispoAlim2017

In [2]:
# Lecture de la table dispoAlim2017
dispoAlim2017 = pd.read_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\DisponibiliteAlimentaire_2017.csv")
dispoAlim2017.head()

Code Domaine                      Domaine  Code zone         Zone  \
0          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
1          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
2          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
3          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
4          FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   

   Code Élément                   Élément  Code Produit          Produit  \
0          5511                Production          2511  Blé et produits   
1          5611   Importations - Quantité          2511  Blé et produits   
2          5072        Variation de stock          2511  Blé et produits   
3          5911   Exportations - Quantité          2511  Blé et produits   
4          5301  Disponibilité intérieure          2511  Blé et produits   

   Code année  Année               Unité  Valeur Symbole  \
0        2017   2017  Milliers de tonnes  4281.0       S   
1        2017   2017  Milliers de tonnes  2302.0       S   
2        2017   2017  Milliers de tonnes  -119.0       S   
3        2017   2017  Milliers de tonnes     0.0       S   
4        2017   2017  Milliers de tonnes  6701.0       S   

  Description du Symbole  
0  Données standardisées  
1  Données standardisées  
2  Données standardisées  
3  Données standardisées  
4  Données standardisées

In [3]:
# Infos et description
display(dispoAlim2017.info())
display(dispoAlim2017.describe())

# Valeurs uniques
print("Nombre de valeurs uniques :")
display(dispoAlim2017.nunique())

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(dispoAlim2017.isnull().sum())

# Doublons
print("Nombre de lignes en doublons :")
display(dispoAlim2017.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 176600 entries, 0 to 176599
Data columns (total 14 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Code Domaine            176600 non-null  object 
 1   Domaine                 176600 non-null  object 
 2   Code zone               176600 non-null  int64  
 3   Zone                    176600 non-null  object 
 4   Code Élément            176600 non-null  int64  
 5   Élément                 176600 non-null  object 
 6   Code Produit            176600 non-null  int64  
 7   Produit                 176600 non-null  object 
 8   Code année              176600 non-null  int64  
 9   Année                   176600 non-null  int64  
 10  Unité                   176600 non-null  object 
 11  Valeur                  176600 non-null  float64
 12  Symbole                 176600 non-null  object 
 13  Description du Symbole  176600 non-null  object 
dtypes: float64(1), int64

None

Code zone   Code Élément   Code Produit  Code année     Année  \
count  176600.000000  176600.000000  176600.000000    176600.0  176600.0   
mean      125.474740    3810.052894    2628.778545      2017.0    2017.0   
std        72.941699    2211.790089      91.993247         0.0       0.0   
min         1.000000     645.000000    2511.000000      2017.0    2017.0   
25%        63.000000     684.000000    2556.000000      2017.0    2017.0   
50%       120.000000    5142.000000    2612.000000      2017.0    2017.0   
75%       188.000000    5301.000000    2732.000000      2017.0    2017.0   
max       276.000000    5911.000000    2899.000000      2017.0    2017.0   

              Valeur  
count  176600.000000  
mean      210.554918  
std      4762.047787  
min    -10388.000000  
25%         0.000000  
50%         0.120000  
75%         8.462500  
max    758548.000000

Nombre de valeurs uniques :


Code Domaine                 1
Domaine                      1
Code zone                  174
Zone                       174
Code Élément                17
Élément                     17
Code Produit                98
Produit                     98
Code année                   1
Année                        1
Unité                        4
Valeur                    7250
Symbole                      2
Description du Symbole       2
dtype: int64

Nombre de valeurs nulles :


Code Domaine              0
Domaine                   0
Code zone                 0
Zone                      0
Code Élément              0
Élément                   0
Code Produit              0
Produit                   0
Code année                0
Année                     0
Unité                     0
Valeur                    0
Symbole                   0
Description du Symbole    0
dtype: int64

Nombre de lignes en doublons :


0

### Colonnes retenues en fonction de l'analyse PESTEL

Les colonnes que nous allons garder pour la construction de notre Dataset suivent la démarche de l'analyse PESTEL.
L'analyse PESTEL est un outil d'analyse stratégique qui permet d'identifier les facteurs macro-environnementaux qui peuvent influencer une entreprise ou une organisation. Les critères de l'analyse PESTEL sont les suivants :

**- Politique** : il s'agit de l'impact des politiques gouvernementales, des réglementations, des lois et des politiques fiscales sur l'entreprise. Cela peut inclure des facteurs tels que les politiques commerciales et les politiques de protection de l'environnement.

**- Économique** : il s'agit de l'impact des conditions économiques sur l'entreprise. Cela peut inclure des facteurs tels que le PIB, le taux de change, le niveau de croissance économique et le taux d'intérêt.

**- Social** : il s'agit de l'impact des changements sociaux sur l'entreprise. Cela peut inclure des facteurs tels que les attitudes et les comportements des consommateurs, les tendances démographiques, la diversité culturelle et les changements de style de vie.

**- Technologique** : il s'agit de l'impact des progrès technologiques sur l'entreprise. Cela peut inclure des facteurs tels que les avancées technologiques, les nouvelles inventions, les nouvelles méthodes de production, l'automatisation et l'impact d'Internet sur les entreprises.

**- Environnemental** : il s'agit de l'impact des facteurs environnementaux sur l'entreprise. Cela peut inclure des facteurs tels que le changement climatique, les catastrophes naturelles, la pollution, les préoccupations environnementales et les réglementations environnementales.

**- Légal** : il s'agit de l'impact des lois et réglementations sur l'entreprise. Cela peut inclure des facteurs tels que les lois sur la santé et la sécurité, les lois du travail, les lois sur la concurrence, les lois sur les droits d'auteur et les lois sur la protection des données.

In [4]:
# Colonnes retenues
dispoAlim2017 = dispoAlim2017[["Zone", "Élément", "Produit", "Année", "Unité", "Valeur"]]
display(dispoAlim2017.head())
dispoAlim2017.shape

Zone                   Élément          Produit  Année  \
0  Afghanistan                Production  Blé et produits   2017   
1  Afghanistan   Importations - Quantité  Blé et produits   2017   
2  Afghanistan        Variation de stock  Blé et produits   2017   
3  Afghanistan   Exportations - Quantité  Blé et produits   2017   
4  Afghanistan  Disponibilité intérieure  Blé et produits   2017   

                Unité  Valeur  
0  Milliers de tonnes  4281.0  
1  Milliers de tonnes  2302.0  
2  Milliers de tonnes  -119.0  
3  Milliers de tonnes     0.0  
4  Milliers de tonnes  6701.0

(176600, 6)

In [5]:
# Affichage des types de produits
dispoAlim2017["Produit"].unique()

array(['Blé et produits', 'Riz et produits', 'Orge et produits',
       'Maïs et produits', 'Seigle et produits', 'Avoine',
       'Millet et produits', 'Sorgho et produits', 'Céréales, Autres',
       'Pommes de Terre et produits', 'Ignames', 'Racines nda',
       'Sucre, canne', 'Sucre, betterave', 'Sucre Eq Brut',
       'Edulcorants Autres', 'Miel', 'Haricots', 'Pois',
       'Légumineuses Autres et produits', 'Noix et produits', 'Soja',
       'Arachides Decortiquees', 'Graines de tournesol',
       'Graines Colza/Moutarde', 'Graines de coton', 'Coco (Incl Coprah)',
       'Sésame', 'Olives', 'Plantes Oleiferes, Autre', 'Huile de Soja',
       "Huile d'Arachide", 'Huile de Tournesol',
       'Huile de Colza&Moutarde', 'Huile Graines de Coton',
       'Huile de Palmistes', 'Huile de Palme', 'Huile de Coco',
       'Huile de Sésame', "Huile d'Olive", 'Huile de Son de Riz',
       'Huile de Germe de Maïs', 'Huil Plantes Oleif Autr',
       'Tomates et produits', 'Oignons', 'Légumes, 

In [6]:
# Conservation des produits de type "Viande de Volailles"
dispoAlim2017 = dispoAlim2017.loc[dispoAlim2017["Produit"]=="Viande de Volailles",:]
display(dispoAlim2017.head())
dispoAlim2017.shape

Zone                   Élément              Produit  Année  \
651  Afghanistan                Production  Viande de Volailles   2017   
652  Afghanistan   Importations - Quantité  Viande de Volailles   2017   
653  Afghanistan        Variation de stock  Viande de Volailles   2017   
654  Afghanistan  Disponibilité intérieure  Viande de Volailles   2017   
655  Afghanistan                    Pertes  Viande de Volailles   2017   

                  Unité  Valeur  
651  Milliers de tonnes    28.0  
652  Milliers de tonnes    29.0  
653  Milliers de tonnes     0.0  
654  Milliers de tonnes    57.0  
655  Milliers de tonnes     2.0

(2061, 6)

In [7]:
# Affichage des éléments
dispoAlim2017["Élément"].unique()

array(['Production', 'Importations - Quantité', 'Variation de stock',
       'Disponibilité intérieure', 'Pertes', 'Résidus', 'Nourriture',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Exportations - Quantité', 'Alimentation pour touristes',
       'Traitement', 'Autres utilisations (non alimentaire)',
       'Aliments pour animaux', 'Semences'], dtype=object)

In [8]:
# Eléments conservés en fonction de l'analyse PESTEL
element = ["Production", "Disponibilité alimentaire (Kcal/personne/jour)"]
dispoAlim2017 = dispoAlim2017.loc[dispoAlim2017["Élément"].isin(element),:]
display(dispoAlim2017.head())
dispoAlim2017.shape

Zone                                         Élément  \
651      Afghanistan                                      Production   
659      Afghanistan  Disponibilité alimentaire (Kcal/personne/jour)   
1704  Afrique du Sud                                      Production   
1714  Afrique du Sud  Disponibilité alimentaire (Kcal/personne/jour)   
2713         Albanie                                      Production   

                  Produit  Année               Unité  Valeur  
651   Viande de Volailles   2017  Milliers de tonnes    28.0  
659   Viande de Volailles   2017  Kcal/personne/jour     5.0  
1704  Viande de Volailles   2017  Milliers de tonnes  1667.0  
1714  Viande de Volailles   2017  Kcal/personne/jour   143.0  
2713  Viande de Volailles   2017  Milliers de tonnes    13.0

(340, 6)

In [9]:
# Pivot de la table dispoAlim2017 selon les éléments
dispoAlim2017_pivot = dispoAlim2017.pivot_table(index="Zone", columns="Élément", values="Valeur", aggfunc=sum).reset_index()
dispoAlim2017_pivot.rename(columns={"Disponibilité alimentaire (Kcal/personne/jour)":"Dispo. alim. volaille (Kcal/pers/jour)", "Production":"Production volaille"}, inplace=True)
display(dispoAlim2017_pivot.head())
display(dispoAlim2017_pivot.shape)

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(dispoAlim2017_pivot.isnull().sum())

Élément            Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
0           Afghanistan                                     5.0   
1        Afrique du Sud                                   143.0   
2               Albanie                                    85.0   
3               Algérie                                    22.0   
4             Allemagne                                    71.0   

Élément  Production volaille  
0                       28.0  
1                     1667.0  
2                       13.0  
3                      275.0  
4                     1514.0

(172, 3)

Nombre de valeurs nulles :


Élément
Zone                                      0
Dispo. alim. volaille (Kcal/pers/jour)    0
Production volaille                       4
dtype: int64

### Table pop2000_2018

In [10]:
# Lecture de la table pop2000_2018
pop2000_2018 = pd.read_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\Population_2000_2018.csv")
display(pop2000_2018.head())
pop2000_2018.shape

Code Domaine                       Domaine  Code zone         Zone  \
0           OA  Séries temporelles annuelles          2  Afghanistan   
1           OA  Séries temporelles annuelles          2  Afghanistan   
2           OA  Séries temporelles annuelles          2  Afghanistan   
3           OA  Séries temporelles annuelles          2  Afghanistan   
4           OA  Séries temporelles annuelles          2  Afghanistan   

   Code Élément            Élément  Code Produit                 Produit  \
0           511  Population totale          3010  Population-Estimations   
1           511  Population totale          3010  Population-Estimations   
2           511  Population totale          3010  Population-Estimations   
3           511  Population totale          3010  Population-Estimations   
4           511  Population totale          3010  Population-Estimations   

   Code année  Année           Unité     Valeur Symbole  \
0        2000   2000  1000 personnes  20779.953       X   
1        2001   2001  1000 personnes  21606.988       X   
2        2002   2002  1000 personnes  22600.770       X   
3        2003   2003  1000 personnes  23680.871       X   
4        2004   2004  1000 personnes  24726.684       X   

          Description du Symbole Note  
0  Sources internationales sûres  NaN  
1  Sources internationales sûres  NaN  
2  Sources internationales sûres  NaN  
3  Sources internationales sûres  NaN  
4  Sources internationales sûres  NaN

(4411, 15)

In [11]:
# Infos et description
display(pop2000_2018.info())
display(pop2000_2018.describe())

# Valeurs uniques
print("Nombre de valeurs uniques :")
display(pop2000_2018.nunique())

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(pop2000_2018.isnull().sum())

# Doublons
print("Nombre de lignes en doublons :")
display(pop2000_2018.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4411 entries, 0 to 4410
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            4411 non-null   object 
 1   Domaine                 4411 non-null   object 
 2   Code zone               4411 non-null   int64  
 3   Zone                    4411 non-null   object 
 4   Code Élément            4411 non-null   int64  
 5   Élément                 4411 non-null   object 
 6   Code Produit            4411 non-null   int64  
 7   Produit                 4411 non-null   object 
 8   Code année              4411 non-null   int64  
 9   Année                   4411 non-null   int64  
 10  Unité                   4411 non-null   object 
 11  Valeur                  4411 non-null   float64
 12  Symbole                 4411 non-null   object 
 13  Description du Symbole  4411 non-null   object 
 14  Note                    258 non-null    

None

Code zone  Code Élément  Code Produit   Code année        Année  \
count  4411.000000        4411.0        4411.0  4411.000000  4411.000000   
mean    132.202902         511.0        3010.0  2009.068238  2009.068238   
std      75.854840           0.0           0.0     5.481539     5.481539   
min       1.000000         511.0        3010.0  2000.000000  2000.000000   
25%      68.000000         511.0        3010.0  2004.000000  2004.000000   
50%     132.000000         511.0        3010.0  2009.000000  2009.000000   
75%     195.000000         511.0        3010.0  2014.000000  2014.000000   
max     299.000000         511.0        3010.0  2018.000000  2018.000000   

             Valeur  
count  4.411000e+03  
mean   2.963004e+04  
std    1.238029e+05  
min    7.850000e-01  
25%    3.921890e+02  
50%    4.764741e+03  
75%    1.821548e+04  
max    1.427648e+06

Nombre de valeurs uniques :


Code Domaine                 1
Domaine                      1
Code zone                  238
Zone                       238
Code Élément                 1
Élément                      1
Code Produit                 1
Produit                      1
Code année                  19
Année                       19
Unité                        1
Valeur                    4398
Symbole                      2
Description du Symbole       2
Note                         1
dtype: int64

Nombre de valeurs nulles :


Code Domaine                 0
Domaine                      0
Code zone                    0
Zone                         0
Code Élément                 0
Élément                      0
Code Produit                 0
Produit                      0
Code année                   0
Année                        0
Unité                        0
Valeur                       0
Symbole                      0
Description du Symbole       0
Note                      4153
dtype: int64

Nombre de lignes en doublons :


0

On compte 238 pays uniques dans la table pop2000_2018.

In [12]:
# Colonnes retenues en fonction de l'analyse PESTEL
pop2000_2018 = pop2000_2018[["Zone", "Année", "Valeur"]].copy()
pop2000_2018 = pop2000_2018.rename(columns={"Valeur":"Population (1000 personnes)"})
display(pop2000_2018.head())
pop2000_2018.shape

Zone  Année  Population (1000 personnes)
0  Afghanistan   2000                    20779.953
1  Afghanistan   2001                    21606.988
2  Afghanistan   2002                    22600.770
3  Afghanistan   2003                    23680.871
4  Afghanistan   2004                    24726.684

(4411, 3)

In [13]:
# Population 2016
pop2016 = pop2000_2018.loc[pop2000_2018["Année"]==2016,:].copy()
pop2016.drop("Année", axis=1, inplace=True)
pop2016.rename(columns={"Population (1000 personnes)":"Population 2016 (pour 1000)"}, inplace=True)
pop2016.head()

Zone  Population 2016 (pour 1000)
16     Afghanistan                    35383.032
35  Afrique du Sud                    56207.646
54         Albanie                     2886.438
73         Algérie                    40551.392
92       Allemagne                    82193.768

In [14]:
# Population 2017
pop2017 = pop2000_2018.loc[pop2000_2018["Année"]==2017,:].copy()
pop2017.drop("Année", axis=1, inplace=True)
pop2017.rename(columns={"Population (1000 personnes)":"Population 2017 (pour 1000)"}, inplace=True)
pop2017.head()

Zone  Population 2017 (pour 1000)
17     Afghanistan                    36296.113
36  Afrique du Sud                    57009.756
55         Albanie                     2884.169
74         Algérie                    41389.189
93       Allemagne                    82658.409

In [15]:
# Jointure des tables pop2016 et pop2017
pop2016_17 = pd.merge(pop2016, pop2017, how="inner", on="Zone")
pop2016_17.head()

Zone  Population 2016 (pour 1000)  Population 2017 (pour 1000)
0     Afghanistan                    35383.032                    36296.113
1  Afrique du Sud                    56207.646                    57009.756
2         Albanie                     2886.438                     2884.169
3         Algérie                    40551.392                    41389.189
4       Allemagne                    82193.768                    82658.409

In [16]:
# %Evolution population 2016/2017
pop2016_17["%Evolution population 2016/2017"] = round(((pop2016_17["Population 2017 (pour 1000)"] - pop2016_17["Population 2016 (pour 1000)"]) / pop2016_17["Population 2016 (pour 1000)"]) * 100 , 2)
pop2016_17.head()

Zone  Population 2016 (pour 1000)  Population 2017 (pour 1000)  \
0     Afghanistan                    35383.032                    36296.113   
1  Afrique du Sud                    56207.646                    57009.756   
2         Albanie                     2886.438                     2884.169   
3         Algérie                    40551.392                    41389.189   
4       Allemagne                    82193.768                    82658.409   

   %Evolution population 2016/2017  
0                             2.58  
1                             1.43  
2                            -0.08  
3                             2.07  
4                             0.57

### Jointure des tables dispoAlim2017 et pop2016_17

In [17]:
# Jointure
data = pd.merge(dispoAlim2017_pivot, pop2016_17[["Zone", "%Evolution population 2016/2017"]], on="Zone" , how="inner")
#data.drop("Année", axis=1, inplace=True)
display(data.head())
display(data.shape)
data.isnull().mean()

Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
0     Afghanistan                                     5.0   
1  Afrique du Sud                                   143.0   
2         Albanie                                    85.0   
3         Algérie                                    22.0   
4       Allemagne                                    71.0   

   Production volaille  %Evolution population 2016/2017  
0                 28.0                             2.58  
1               1667.0                             1.43  
2                 13.0                            -0.08  
3                275.0                             2.07  
4               1514.0                             0.57

(172, 4)

Zone                                      0.000000
Dispo. alim. volaille (Kcal/pers/jour)    0.000000
Production volaille                       0.023256
%Evolution population 2016/2017           0.000000
dtype: float64

### Table population_granularité_2017

In [18]:
# Lecture de la table bilans_alimentaires_2017
population_granularité_2017 = pd.read_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\population_granularité_2017.csv")
display(population_granularité_2017.head())
population_granularité_2017.shape

Code Domaine                       Domaine  Code zone (M49)         Zone  \
0           OA  Séries temporelles annuelles                4  Afghanistan   
1           OA  Séries temporelles annuelles                4  Afghanistan   
2           OA  Séries temporelles annuelles                4  Afghanistan   
3           OA  Séries temporelles annuelles                4  Afghanistan   
4           OA  Séries temporelles annuelles                4  Afghanistan   

   Code Élément             Élément  Code Produit                 Produit  \
0           511   Population totale          3010  Population-Estimations   
1           512              Hommes          3010  Population-Estimations   
2           513              Femmes          3010  Population-Estimations   
3           551   Population rurale          3010  Population-Estimations   
4           561  Population urbaine          3010  Population-Estimations   

   Code année  Année           Unité     Valeur Symbole  \
0        2017   2017  1000 personnes  35643.418       X   
1        2017   2017  1000 personnes  18028.696       X   
2        2017   2017  1000 personnes  17614.722       X   
3        2017   2017  1000 personnes  26558.609       X   
4        2017   2017  1000 personnes   8971.472       X   

              Description du Symbole  Note  
0  Ciffre de sources internationales   NaN  
1  Ciffre de sources internationales   NaN  
2  Ciffre de sources internationales   NaN  
3  Ciffre de sources internationales   NaN  
4  Ciffre de sources internationales   NaN

(1173, 15)

In [19]:
# Infos et description
display(population_granularité_2017.info())
display(population_granularité_2017.describe())

# Valeurs uniques
print("Nombre de valeurs uniques :")
display(population_granularité_2017.nunique())

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(population_granularité_2017.isnull().sum())

# Doublons
print("Nombre de lignes en doublons :")
display(population_granularité_2017.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1173 entries, 0 to 1172
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            1173 non-null   object 
 1   Domaine                 1173 non-null   object 
 2   Code zone (M49)         1173 non-null   int64  
 3   Zone                    1173 non-null   object 
 4   Code Élément            1173 non-null   int64  
 5   Élément                 1173 non-null   object 
 6   Code Produit            1173 non-null   int64  
 7   Produit                 1173 non-null   object 
 8   Code année              1173 non-null   int64  
 9   Année                   1173 non-null   int64  
 10  Unité                   1173 non-null   object 
 11  Valeur                  1173 non-null   float64
 12  Symbole                 1173 non-null   object 
 13  Description du Symbole  1173 non-null   object 
 14  Note                    0 non-null      

None

Code zone (M49)  Code Élément  Code Produit  Code année   Année  \
count      1173.000000   1173.000000        1173.0      1173.0  1173.0   
mean        435.792839    529.554987        3010.0      2017.0  2017.0   
std         251.235898     21.794998           0.0         0.0     0.0   
min           4.000000    511.000000        3010.0      2017.0  2017.0   
25%         218.000000    512.000000        3010.0      2017.0  2017.0   
50%         438.000000    513.000000        3010.0      2017.0  2017.0   
75%         654.000000    551.000000        3010.0      2017.0  2017.0   
max         894.000000    561.000000        3010.0      2017.0  2017.0   

             Valeur  Note  
count  1.173000e+03   0.0  
mean   2.307871e+04   NaN  
std    1.029616e+05   NaN  
min    0.000000e+00   NaN  
25%    2.076300e+02   NaN  
50%    2.813212e+03   NaN  
75%    1.138449e+04   NaN  
max    1.442041e+06   NaN

Nombre de valeurs uniques :


Code Domaine                 1
Domaine                      1
Code zone (M49)            236
Zone                       236
Code Élément                 5
Élément                      5
Code Produit                 1
Produit                      1
Code année                   1
Année                        1
Unité                        1
Valeur                    1160
Symbole                      1
Description du Symbole       1
Note                         0
dtype: int64

Nombre de valeurs nulles :


Code Domaine                 0
Domaine                      0
Code zone (M49)              0
Zone                         0
Code Élément                 0
Élément                      0
Code Produit                 0
Produit                      0
Code année                   0
Année                        0
Unité                        0
Valeur                       0
Symbole                      0
Description du Symbole       0
Note                      1173
dtype: int64

Nombre de lignes en doublons :


0

In [20]:
# Colonnes retenues
population_granularité_2017 = population_granularité_2017[["Zone", "Élément", "Valeur"]]
display(population_granularité_2017.head())
population_granularité_2017.shape

Zone             Élément     Valeur
0  Afghanistan   Population totale  35643.418
1  Afghanistan              Hommes  18028.696
2  Afghanistan              Femmes  17614.722
3  Afghanistan   Population rurale  26558.609
4  Afghanistan  Population urbaine   8971.472

(1173, 3)

In [21]:
# Eléments uniques
population_granularité_2017["Élément"].unique()

array(['Population totale', 'Hommes', 'Femmes', 'Population rurale',
       'Population urbaine'], dtype=object)

In [22]:
# Eléments conservés en fonction de l'analyse PESTEL
population_granularité_2017 = population_granularité_2017.loc[population_granularité_2017["Élément"]=="Population totale",:]
display(population_granularité_2017.head())
population_granularité_2017.shape

Zone            Élément     Valeur
0      Afghanistan  Population totale  35643.418
5   Afrique du Sud  Population totale  56641.209
10         Albanie  Population totale   2879.355
15         Algérie  Population totale  41136.546
20       Allemagne  Population totale  82624.374

(235, 3)

In [23]:
# Pivot de la table population_granularité_2017
population_granularité_2017_pivot = population_granularité_2017.pivot_table(index="Zone", columns="Élément", values="Valeur", aggfunc=sum).reset_index()
display(population_granularité_2017_pivot.head())
population_granularité_2017_pivot.shape

Élément            Zone  Population totale
0           Afghanistan          35643.418
1        Afrique du Sud          56641.209
2               Albanie           2879.355
3               Algérie          41136.546
4             Allemagne          82624.374

(235, 2)

### Jointure de la table population_granularité_2017 avec data

In [24]:
# Jointure
data = pd.merge(data, population_granularité_2017_pivot, on="Zone", how="inner")
data.rename(columns={"Population totale":"Population totale (1000 pers.)"}, inplace=True)
display(data.head())
display(data.shape)
data.isnull().mean()

Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
0     Afghanistan                                     5.0   
1  Afrique du Sud                                   143.0   
2         Albanie                                    85.0   
3         Algérie                                    22.0   
4       Allemagne                                    71.0   

   Production volaille  %Evolution population 2016/2017  \
0                 28.0                             2.58   
1               1667.0                             1.43   
2                 13.0                            -0.08   
3                275.0                             2.07   
4               1514.0                             0.57   

   Population totale (1000 pers.)  
0                       35643.418  
1                       56641.209  
2                        2879.355  
3                       41136.546  
4                       82624.374

(172, 5)

Zone                                      0.000000
Dispo. alim. volaille (Kcal/pers/jour)    0.000000
Production volaille                       0.023256
%Evolution population 2016/2017           0.000000
Population totale (1000 pers.)            0.000000
dtype: float64

### Table sécurité_alimentaire_2017

In [25]:
# Lecture de la table sécurité_alimentaire_2017
sécurité_alimentaire_2017 = pd.read_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\sécurité_alimentaire_2017.csv")
display(sécurité_alimentaire_2017.head())
sécurité_alimentaire_2017.shape

Code Domaine                             Domaine  Code zone (M49)  \
0           FS  Données de la sécurité alimentaire                4   
1           FS  Données de la sécurité alimentaire                4   
2           FS  Données de la sécurité alimentaire                4   
3           FS  Données de la sécurité alimentaire                4   
4           FS  Données de la sécurité alimentaire                4   

          Zone  Code Élément Élément  Code Produit  \
0  Afghanistan          6121  Valeur         21010   
1  Afghanistan          6128  Valeur         22000   
2  Afghanistan          6123  Valeur         21013   
3  Afghanistan          6123  Valeur         21014   
4  Afghanistan          6126  Valeur         22013   

                                             Produit  Code année      Année  \
0  Suffisance des apports énergétiques alimentair...    20162018  2016-2018   
1  Disponibilité alimentaire par habitant utilisé...    20162018  2016-2018   
2  Disponibilités protéiques moyennes (g/personne...    20162018  2016-2018   
3  Disponibilités protéines moyennes d’origine an...    20162018  2016-2018   
4  PIB par habitant, ($ PPA internationaux consta...        2017       2017   

                Unité  Valeur Symbole             Description du Symbole  Note  
0                   %   106.0       E                     Valeur estimée   NaN  
1  Kcal/personne/jour  2266.0       E                     Valeur estimée   NaN  
2     g/personne/jour    58.0       E                     Valeur estimée   NaN  
3     g/personne/jour    10.7       E                     Valeur estimée   NaN  
4                  I$  2058.4       X  Ciffre de sources internationales   NaN

(3061, 15)

In [26]:
# Infos et description
display(sécurité_alimentaire_2017.info())
display(sécurité_alimentaire_2017.describe())

# Valeurs uniques
print("Nombre de valeurs uniques :")
display(sécurité_alimentaire_2017.nunique())

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(sécurité_alimentaire_2017.isnull().sum())

# Doublons
print("Nombre de lignes en doublons :")
display(sécurité_alimentaire_2017.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3061 entries, 0 to 3060
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            3061 non-null   object 
 1   Domaine                 3061 non-null   object 
 2   Code zone (M49)         3061 non-null   int64  
 3   Zone                    3061 non-null   object 
 4   Code Élément            3061 non-null   int64  
 5   Élément                 3061 non-null   object 
 6   Code Produit            3061 non-null   int64  
 7   Produit                 3061 non-null   object 
 8   Code année              3061 non-null   int64  
 9   Année                   3061 non-null   object 
 10  Unité                   3061 non-null   object 
 11  Valeur                  2815 non-null   float64
 12  Symbole                 3061 non-null   object 
 13  Description du Symbole  3061 non-null   object 
 14  Note                    0 non-null      

None

Code zone (M49)  Code Élément  Code Produit    Code année  \
count      3061.000000   3061.000000   3061.000000  3.061000e+03   
mean        426.607971   6127.560928  21160.443646  6.719822e+06   
std         252.992246     12.480572    324.075630  9.504292e+06   
min           4.000000   6121.000000  21010.000000  2.017000e+03   
25%         208.000000   6121.000000  21030.000000  2.017000e+03   
50%         422.000000   6125.000000  21045.000000  2.017000e+03   
75%         642.000000   6128.000000  21057.000000  2.016202e+07   
max         894.000000   6173.000000  22013.000000  2.016202e+07   

              Valeur  Note  
count    2815.000000   0.0  
mean     1875.786451   NaN  
std      7676.058760   NaN  
min        -2.940000   NaN  
25%        17.000000   NaN  
50%        79.800000   NaN  
75%      1694.500000   NaN  
max    126144.000000   NaN

Nombre de valeurs uniques :


Code Domaine                 1
Domaine                      1
Code zone (M49)            204
Zone                       204
Code Élément                 7
Élément                      1
Code Produit                16
Produit                     16
Code année                   2
Année                        2
Unité                        7
Valeur                    1447
Symbole                      3
Description du Symbole       3
Note                         0
dtype: int64

Nombre de valeurs nulles :


Code Domaine                 0
Domaine                      0
Code zone (M49)              0
Zone                         0
Code Élément                 0
Élément                      0
Code Produit                 0
Produit                      0
Code année                   0
Année                        0
Unité                        0
Valeur                     246
Symbole                      0
Description du Symbole       0
Note                      3061
dtype: int64

Nombre de lignes en doublons :


0

In [27]:
# Affichage des valeurs nulles
sécurité_alimentaire_2017.loc[sécurité_alimentaire_2017["Valeur"].isnull(),:].head()

Code Domaine                             Domaine  Code zone (M49)     Zone  \
77           FS  Données de la sécurité alimentaire               20  Andorre   
78           FS  Données de la sécurité alimentaire               20  Andorre   
79           FS  Données de la sécurité alimentaire               20  Andorre   
80           FS  Données de la sécurité alimentaire               20  Andorre   
81           FS  Données de la sécurité alimentaire               20  Andorre   

    Code Élément Élément  Code Produit  \
77          6121  Valeur         21010   
78          6128  Valeur         22000   
79          6123  Valeur         21013   
80          6123  Valeur         21014   
81          6121  Valeur         21033   

                                              Produit  Code année      Année  \
77  Suffisance des apports énergétiques alimentair...    20162018  2016-2018   
78  Disponibilité alimentaire par habitant utilisé...    20162018  2016-2018   
79  Disponibilités protéiques moyennes (g/personne...    20162018  2016-2018   
80  Disponibilités protéines moyennes d’origine an...    20162018  2016-2018   
81  Valeur des importations alimentaires par rappo...    20162018  2016-2018   

                 Unité  Valeur Symbole Description du Symbole  Note  
77                   %     NaN       O       Valeur manquante   NaN  
78  Kcal/personne/jour     NaN       O       Valeur manquante   NaN  
79     g/personne/jour     NaN       O       Valeur manquante   NaN  
80     g/personne/jour     NaN       O       Valeur manquante   NaN  
81                   %     NaN       O       Valeur manquante   NaN

Il s'agit de valeurs non communiquées par l'organisme de collecte des données, sous la notation "Valeur manquante".

In [28]:
# Eléments uniques dans la colonne Produit
sécurité_alimentaire_2017["Produit"].unique()

array(['Suffisance des apports énergétiques alimentaires moyens (%) (moyenne sur 3 ans)',
       "Disponibilité alimentaire par habitant utilisée dans l'estimation de la prévalence de la sous-alimentation (kcal/personne/jour) (moyenne sur 3 ans)",
       'Disponibilités protéiques moyennes (g/personne/jour) (moyenne sur 3 ans)',
       'Disponibilités protéines moyennes d’origine animale (g/personne/jour) (moyenne sur 3 ans)',
       'PIB par habitant, ($ PPA internationaux constants de 2017)',
       'Valeur des importations alimentaires par rapport aux exportations totales de marchandises (%) (moyenne sur 3 ans)',
       'Stabilité politique et absence de violence/terrorisme (indice)',
       'Variabilité de la production alimentaire par habitant ($I par personne constant 2014-16)',
       'Variabilité des disponibilités alimentaires par habitant (Kcal/personne/jour)',
       "Pourcentage de la population ayant accès à des services d'eau potable gérés en toute sécurité",
       "Pour

In [29]:
# Eléments conservés en fonction de l'analyse PESTEL
sécurité_alimentaire_2017 = sécurité_alimentaire_2017.loc[sécurité_alimentaire_2017["Produit"]=="Stabilité politique et absence de violence/terrorisme (indice)",:]
display(sécurité_alimentaire_2017.head())
sécurité_alimentaire_2017.shape

Code Domaine                             Domaine  Code zone (M49)  \
6            FS  Données de la sécurité alimentaire                4   
21           FS  Données de la sécurité alimentaire              710   
35           FS  Données de la sécurité alimentaire                8   
51           FS  Données de la sécurité alimentaire               12   
67           FS  Données de la sécurité alimentaire              276   

              Zone  Code Élément Élément  Code Produit  \
6      Afghanistan          6125  Valeur         21032   
21  Afrique du Sud          6125  Valeur         21032   
35         Albanie          6125  Valeur         21032   
51         Algérie          6125  Valeur         21032   
67       Allemagne          6125  Valeur         21032   

                                              Produit  Code année Année  \
6   Stabilité politique et absence de violence/ter...        2017  2017   
21  Stabilité politique et absence de violence/ter...        2017  2017   
35  Stabilité politique et absence de violence/ter...        2017  2017   
51  Stabilité politique et absence de violence/ter...        2017  2017   
67  Stabilité politique et absence de violence/ter...        2017  2017   

     Unité  Valeur Symbole             Description du Symbole  Note  
6   indice   -2.80       X  Ciffre de sources internationales   NaN  
21  indice   -0.28       X  Ciffre de sources internationales   NaN  
35  indice    0.38       X  Ciffre de sources internationales   NaN  
51  indice   -0.92       X  Ciffre de sources internationales   NaN  
67  indice    0.59       X  Ciffre de sources internationales   NaN

(197, 15)

In [30]:
# Colonnes retenues
sécurité_alimentaire_2017 = sécurité_alimentaire_2017[["Zone", "Produit", "Valeur", "Unité"]]
display(sécurité_alimentaire_2017.head())
sécurité_alimentaire_2017.shape

Zone                                            Produit  Valeur  \
6      Afghanistan  Stabilité politique et absence de violence/ter...   -2.80   
21  Afrique du Sud  Stabilité politique et absence de violence/ter...   -0.28   
35         Albanie  Stabilité politique et absence de violence/ter...    0.38   
51         Algérie  Stabilité politique et absence de violence/ter...   -0.92   
67       Allemagne  Stabilité politique et absence de violence/ter...    0.59   

     Unité  
6   indice  
21  indice  
35  indice  
51  indice  
67  indice

(197, 4)

In [45]:
# Pivot de la table sécurité_alimentaire_2017
sécurité_alimentaire_2017_pivot = sécurité_alimentaire_2017.pivot_table(index="Zone", columns="Produit", values="Valeur", aggfunc=sum).reset_index()
sécurité_alimentaire_2017_pivot.rename(columns={"Besoins énergétiques alimentaires moyens (kcal/personne/jour)":"Besoins énerg. alim. moyens (kcal/pers./jour)", "Stabilité politique et absence de violence/terrorisme (indice)":"Stabilité politique (indice)"}, inplace=True)
display(sécurité_alimentaire_2017_pivot.head())
display(sécurité_alimentaire_2017_pivot.shape)

# Moyenne des valeurs nulles (en dataframe)
pd.DataFrame(sécurité_alimentaire_2017_pivot.isnull().mean()).rename(columns={0:"null_mean"})

Produit            Zone  Stabilité politique (indice)
0           Afghanistan                         -2.80
1        Afrique du Sud                         -0.28
2               Albanie                          0.38
3               Algérie                         -0.92
4             Allemagne                          0.59

(197, 2)

null_mean
Produit                                
Zone                                0.0
Stabilité politique (indice)        0.0

### Jointure de la table sécurité_alimentaire_2017 avec data

In [32]:
# Jointure
data = pd.merge(data, sécurité_alimentaire_2017_pivot, on="Zone", how="inner")
display(data.head())
display(data.shape)
pd.DataFrame(data.isnull().mean()).rename(columns={0:"null_mean"})

Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
0     Afghanistan                                     5.0   
1  Afrique du Sud                                   143.0   
2         Albanie                                    85.0   
3         Algérie                                    22.0   
4       Allemagne                                    71.0   

   Production volaille  %Evolution population 2016/2017  \
0                 28.0                             2.58   
1               1667.0                             1.43   
2                 13.0                            -0.08   
3                275.0                             2.07   
4               1514.0                             0.57   

   Population totale (1000 pers.)  Stabilité politique (indice)  
0                       35643.418                         -2.80  
1                       56641.209                         -0.28  
2                        2879.355                          0.38  
3                       41136.546                         -0.92  
4                       82624.374                          0.59

(169, 6)

null_mean
Zone                                     0.000000
Dispo. alim. volaille (Kcal/pers/jour)   0.000000
Production volaille                      0.023669
%Evolution population 2016/2017          0.000000
Population totale (1000 pers.)           0.000000
Stabilité politique (indice)             0.000000

### Table indicateurs_macro_2017

In [33]:
# Lecture de la table indicateurs_macro_2017
indicateurs_macro_2017 = pd.read_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\indicateurs_macro_2017.csv")
display(indicateurs_macro_2017.head())
indicateurs_macro_2017.shape

Code Domaine            Domaine  Code zone (M49)            Zone  \
0           MK  Indicateurs macro                4     Afghanistan   
1           MK  Indicateurs macro                4     Afghanistan   
2           MK  Indicateurs macro              710  Afrique du Sud   
3           MK  Indicateurs macro              710  Afrique du Sud   
4           MK  Indicateurs macro                8         Albanie   

   Code Élément                   Élément  Code Produit  \
0          6110               Valeur US $         22008   
1          6119  Valeur US $ par habitant         22008   
2          6110               Valeur US $         22008   
3          6119  Valeur US $ par habitant         22008   
4          6110               Valeur US $         22008   

                  Produit  Code année  Année     Unité         Valeur Symbole  \
0  Produit Intérieur Brut        2017   2017  millions   18896.352020       X   
1  Produit Intérieur Brut        2017   2017       US$     530.149831       X   
2  Produit Intérieur Brut        2017   2017  millions  380851.444100       X   
3  Produit Intérieur Brut        2017   2017       US$    6723.928582       X   
4  Produit Intérieur Brut        2017   2017  millions   13019.729860       X   

              Description du Symbole  Note  
0  Ciffre de sources internationales   NaN  
1  Ciffre de sources internationales   NaN  
2  Ciffre de sources internationales   NaN  
3  Ciffre de sources internationales   NaN  
4  Ciffre de sources internationales   NaN

(422, 15)

In [34]:
# Infos et description
display(indicateurs_macro_2017.info())
display(indicateurs_macro_2017.describe())

# Valeurs uniques
print("Nombre de valeurs uniques :")
display(indicateurs_macro_2017.nunique())

# Valeurs nulles
print("Nombre de valeurs nulles :")
display(indicateurs_macro_2017.isnull().sum())

# Doublons
print("Nombre de lignes en doublons :")
display(indicateurs_macro_2017.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 422 entries, 0 to 421
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            422 non-null    object 
 1   Domaine                 422 non-null    object 
 2   Code zone (M49)         422 non-null    int64  
 3   Zone                    422 non-null    object 
 4   Code Élément            422 non-null    int64  
 5   Élément                 422 non-null    object 
 6   Code Produit            422 non-null    int64  
 7   Produit                 422 non-null    object 
 8   Code année              422 non-null    int64  
 9   Année                   422 non-null    int64  
 10  Unité                   422 non-null    object 
 11  Valeur                  422 non-null    float64
 12  Symbole                 422 non-null    object 
 13  Description du Symbole  422 non-null    object 
 14  Note                    0 non-null      fl

None

Code zone (M49)  Code Élément  Code Produit  Code année   Année  \
count       422.000000    422.000000         422.0       422.0   422.0   
mean        429.407583   6114.500000       22008.0      2017.0  2017.0   
std         250.939413      4.505341           0.0         0.0     0.0   
min           4.000000   6110.000000       22008.0      2017.0  2017.0   
25%         209.000000   6110.000000       22008.0      2017.0  2017.0   
50%         430.000000   6114.500000       22008.0      2017.0  2017.0   
75%         642.750000   6119.000000       22008.0      2017.0  2017.0   
max         894.000000   6119.000000       22008.0      2017.0  2017.0   

             Valeur  Note  
count  4.220000e+02   0.0  
mean   2.298944e+05   NaN  
std    1.341679e+06   NaN  
min    4.529575e+01   NaN  
25%    3.081183e+03   NaN  
50%    1.269970e+04   NaN  
75%    4.855695e+04   NaN  
max    1.947734e+07   NaN

Nombre de valeurs uniques :


Code Domaine                1
Domaine                     1
Code zone (M49)           211
Zone                      211
Code Élément                2
Élément                     2
Code Produit                1
Produit                     1
Code année                  1
Année                       1
Unité                       2
Valeur                    422
Symbole                     1
Description du Symbole      1
Note                        0
dtype: int64

Nombre de valeurs nulles :


Code Domaine                0
Domaine                     0
Code zone (M49)             0
Zone                        0
Code Élément                0
Élément                     0
Code Produit                0
Produit                     0
Code année                  0
Année                       0
Unité                       0
Valeur                      0
Symbole                     0
Description du Symbole      0
Note                      422
dtype: int64

Nombre de lignes en doublons :


0

In [35]:
# Colonnes retenues
indicateurs_macro_2017 = indicateurs_macro_2017[["Zone", "Élément", "Produit", "Valeur", "Unité"]]
display(indicateurs_macro_2017.head())
indicateurs_macro_2017.shape

Zone                   Élément                 Produit  \
0     Afghanistan               Valeur US $  Produit Intérieur Brut   
1     Afghanistan  Valeur US $ par habitant  Produit Intérieur Brut   
2  Afrique du Sud               Valeur US $  Produit Intérieur Brut   
3  Afrique du Sud  Valeur US $ par habitant  Produit Intérieur Brut   
4         Albanie               Valeur US $  Produit Intérieur Brut   

          Valeur     Unité  
0   18896.352020  millions  
1     530.149831       US$  
2  380851.444100  millions  
3    6723.928582       US$  
4   13019.729860  millions

(422, 5)

In [36]:
# Valeurs uniques dans la colonne Produit
indicateurs_macro_2017["Produit"].unique()

array(['Produit Intérieur Brut'], dtype=object)

In [37]:
# Valeurs uniques dans la colonne Élément (Tous les éléments sont conservés pour l'analyse PESTEL)
indicateurs_macro_2017["Élément"].unique()

array(['Valeur US $', 'Valeur US $ par habitant'], dtype=object)

In [38]:
# Pivot de la table indicateurs_macro_2017
indicateurs_macro_2017_pivot = indicateurs_macro_2017.pivot_table(index="Zone", columns="Élément", values="Valeur", aggfunc=sum).reset_index()
indicateurs_macro_2017_pivot.rename(columns={"Valeur US $":"PIB US $", "Valeur US $ par habitant":"PIB US $ par habitant"}, inplace=True)
display(indicateurs_macro_2017_pivot.head())
indicateurs_macro_2017_pivot.shape

Élément            Zone      PIB US $  PIB US $ par habitant
0           Afghanistan  1.889635e+04             530.149831
1        Afrique du Sud  3.808514e+05            6723.928582
2               Albanie  1.301973e+04            4521.752219
3               Algérie  1.700970e+05            4134.936054
4             Allemagne  3.690849e+06           44670.222288

(211, 3)

### Jointure de la table indicateurs_macro_2017 avec data

In [46]:
# Jointure
data = pd.merge(data, indicateurs_macro_2017_pivot, on="Zone", how="inner")
display(data.head())
display(data.shape)
pd.DataFrame(data.isnull().mean()).rename(columns={0:"null_mean"})

Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
0     Afghanistan                                     5.0   
1  Afrique du Sud                                   143.0   
2         Albanie                                    85.0   
3         Algérie                                    22.0   
4       Allemagne                                    71.0   

   Production volaille  %Evolution population 2016/2017  \
0                 28.0                             2.58   
1               1667.0                             1.43   
2                 13.0                            -0.08   
3                275.0                             2.07   
4               1514.0                             0.57   

   Population totale (1000 pers.)  Stabilité politique (indice)    PIB US $_x  \
0                       35643.418                         -2.80  1.889635e+04   
1                       56641.209                         -0.28  3.808514e+05   
2                        2879.355                          0.38  1.301973e+04   
3                       41136.546                         -0.92  1.700970e+05   
4                       82624.374                          0.59  3.690849e+06   

   PIB US $ par habitant_x    PIB US $_y  PIB US $ par habitant_y  
0               530.149831  1.889635e+04               530.149831  
1              6723.928582  3.808514e+05              6723.928582  
2              4521.752219  1.301973e+04              4521.752219  
3              4134.936054  1.700970e+05              4134.936054  
4             44670.222288  3.690849e+06             44670.222288

(168, 10)

null_mean
Zone                                          0.0
Dispo. alim. volaille (Kcal/pers/jour)        0.0
Production volaille                           0.0
%Evolution population 2016/2017               0.0
Population totale (1000 pers.)                0.0
Stabilité politique (indice)                  0.0
PIB US $_x                                    0.0
PIB US $ par habitant_x                       0.0
PIB US $_y                                    0.0
PIB US $ par habitant_y                       0.0

In [40]:
# Remplacement des valeurs manquantes par la moyenne des valeurs de chaque colonne
data["Production volaille"] = data["Production volaille"].fillna(data["Production volaille"].mean())
data["Stabilité politique (indice)"] = data["Stabilité politique (indice)"].fillna(data["Stabilité politique (indice)"].mean())

pd.DataFrame(data.isnull().mean()).rename(columns={0:"null_mean"})

null_mean
Zone                                          0.0
Dispo. alim. volaille (Kcal/pers/jour)        0.0
Production volaille                           0.0
%Evolution population 2016/2017               0.0
Population totale (1000 pers.)                0.0
Stabilité politique (indice)                  0.0
PIB US $                                      0.0
PIB US $ par habitant                         0.0

In [41]:
# Liste des pays pour identifier les outliers (Les outliers seront détectés dans l'analyse sur le second notebook)
data["Zone"].unique()

array(['Afghanistan', 'Afrique du Sud', 'Albanie', 'Algérie', 'Allemagne',
       'Angola', 'Antigua-et-Barbuda', 'Arabie saoudite', 'Argentine',
       'Arménie', 'Australie', 'Autriche', 'Azerbaïdjan', 'Bahamas',
       'Bangladesh', 'Barbade', 'Belgique', 'Belize',
       'Bolivie (État plurinational de)', 'Bosnie-Herzégovine',
       'Botswana', 'Brésil', 'Bulgarie', 'Burkina Faso', 'Bélarus',
       'Bénin', 'Cabo Verde', 'Cambodge', 'Cameroun', 'Canada', 'Chili',
       'Chine - RAS de Hong-Kong', 'Chine - RAS de Macao', 'Chypre',
       'Colombie', 'Congo', 'Costa Rica', 'Croatie', 'Cuba',
       "Côte d'Ivoire", 'Danemark', 'Djibouti', 'Dominique',
       'El Salvador', 'Espagne', 'Estonie', 'Eswatini', 'Fidji',
       'Finlande', 'France', 'Fédération de Russie', 'Gabon', 'Gambie',
       'Ghana', 'Grenade', 'Grèce', 'Guatemala', 'Guinée',
       'Guinée-Bissau', 'Guyana', 'Géorgie', 'Haïti', 'Honduras',
       'Hongrie', 'Inde', 'Indonésie', "Iran (République islamique d')",


In [42]:
# Séparation des outliers dans un autre tableau (Outliers détectés dans le second notebook)
outliers = ["Brésil", "Inde", "États-Unis d'Amérique"]
data1 = data.loc[~data["Zone"].isin(outliers),:]
data2 = data.loc[data["Zone"].isin(outliers),:]
data2

Zone  Dispo. alim. volaille (Kcal/pers/jour)  \
21                  Brésil                                   204.0   
64                    Inde                                     7.0   
165  États-Unis d'Amérique                                   219.0   

     Production volaille  %Evolution population 2016/2017  \
21               14201.0                             0.81   
64                3545.0                             1.07   
165              21914.0                             0.64   

     Population totale (1000 pers.)  Stabilité politique (indice)  \
21                       208504.960                         -0.47   
64                      1354195.680                         -0.76   
165                      329791.231                          0.29   

         PIB US $  PIB US $ par habitant  
21   2.063515e+06            9896.717512  
64   2.624329e+06            1937.924543  
165  1.947734e+07           59059.596403

### Exportation des fichiers CSV

In [43]:
data.to_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\dataset_p9.csv", index=False)
data1.to_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\dataset1_p9.csv", index=False)
data2.to_csv("C:\\Users\open_\P9_Produisez_une_étude_de_marché\dataset2_p9.csv", index=False)